news/2017-01-19-bean-validation-2-0-progress-report.adoc

```diff
@@ -0,0 +1,235 @@
+----
+@ZipCode(countryCode = "fr", groups = Default.class, message = "zip code is not valid")
+@ZipCode(
+    countryCode = "fr",
+    groups = SuperUser.class,
+    message = "zip code invalid. Requires overriding before saving."
+)
+private String zipCode;
+----
+
+`ConstraintValidator#initialize()` has an empty default implementation now (https://hibernate.atlassian.net/browse/BVAL-555[BVAL-555]),
+simplifying the implementation of constraint validators that don't need to access any constraint attributes.
+You can simply omit the `initialize()` method:
+
+[source,java]
+----
+public class AssertTrueValidator implements ConstraintValidator<AssertTrue, Boolean> {
+
+    @Override
+    public boolean isValid(Boolean bool, ConstraintValidatorContext constraintValidatorContext) {
+        return bool == null || bool;
+    }
+}
+----
+
+Another nice improvement is the usage of actual parameter names when reporting constraint violations for constraints on method or constructor parameters (https://hibernate.atlassian.net/browse/BVAL-498[BVAL-498]).
+Provided you have enabled reflective parameter name access during compilation (using `-parameters` javac option),
+`Path.Node#getName()` will return the actual parameter name instead of "arg0", "arg1" for parameter nodes.
+
+== Next steps
+
+We are currently working on picking up some loose ends around the proposal for container element validation.
+Once that is done, we feel it is the right time to put out an Alpha1 release of Bean Validation 2.0 and post it for Early Draft Review to the JCP.
+This should get the discussed changes into the hands of more people out there and will let us improve and hone the features added so far.
+
+In parallel we'll continue with some other features from the backlog.
+Issues high on our priority list are:
+
+* Adding some new constraints as per our link:/news/2016/09/15/which-constraints-to-add/[recent survey], e.g. `@NotEmpty`, `@NotBlank`
+* Separating the notions of message resolver and message interpolator (https://hibernate.atlassian.net/browse/BVAL-217[BVAL-217])
+* Ability to validate an object and a list of changes (https://hibernate.atlassian.net/browse/BVAL-214[BVAL-214])
+
+We also contemplate the idea of using Java 8 Lambda expressions and method references for defining constraints without an explicit `ConstraintValidator` implementation class.
+This is already supported in the reference implementation:
+
+[source,java]
+----
+ConstraintMapping mapping = ...
+mapping.constraintDefinition( Directory.class ) // @Directory is a constraint annotation
+    .validateType( File.class ).with( File::exists );
+----
+
+We haven't decided yet whether to put this into the spec or not.
+So we recommend you give it a try in the reference implementation and let us know about your thoughts.
+The feedback when https://twitter.com/gunnarmorling/status/819631488358563840[sharing the idea] on Twitter was https://twitter.com/dblevins/status/819633752888475648[very encouraging].
```

Comment:
You know David has been pushing for it in the first place so that's not surprising